src/bsoncxx/document/value.cpp

```diff
@@ -53,11 +54,7 @@ value& value::operator=(const value& rhs) {
 }
 
 value::unique_ptr_type value::release() {
-    auto x = std::move(_data);
-
-    _data.release();
-
-    return x;
+    return std::move(_data);
```

Comment:
It is required, because `_data` is a class member.  It's not necessary for `x` because that's a local variable which gets copy-elision (or in the absence of copy-elision support, the return checks first for a constructor consuming an rvalue before one consuming an lvalue).